src/Confluent.Kafka/Timestamp.cs

```diff
@@ -36,16 +36,38 @@ public struct Timestamp
         /// <summary>
         ///     Initializes a new instance of the Timestamp structure.
         /// </summary>
+        /// <param name="unixTimestampMs">
+        ///     The unix millisecond timestamp.
+        /// </param>
+        /// <param name="type">
+        ///     The type of the timestamp.
+        /// </param>
+        public Timestamp(long unixTimestampMs, TimestampType type)
+        {
+            Type = type;
+            UnixTimestampMs = unixTimestampMs;
+        }
+
+        /// <summary>
+        ///     Initializes a new instance of the Timestamp structure.
+        ///     Prefer the (long, TimestampType) overload if possible.
```

Comment:
actually, since there was a bug that made all Timstamp values incorrect in the released version, I think the DateTime property could just be removed (i.e. skip depreciating it).